### PR TITLE
#244 - add change shop/ upgrade tree toggle

### PR DESCRIPTION
**Pull request by SFarazH**
Solved Issue #244 
Added a toggle switch instead of the change shop / upgrade tree button

### CODE REVIEW

File: core/development/ui/Events.ts [hunk 1]
```diff
@@ -51,7 +51,8 @@ export function initializeEventListeners() {
   HELL_BOARD_BUTTON?.addEventListener('click', handleBoardButtonPress);
   HEAVEN_BOARD_BUTTON?.addEventListener('click', handleBoardButtonPress);
 
-  SHOP_UPGRADE_SWAPPER?.addEventListener('click', swapShopAndUpgrade);
+  // SHOP_UPGRADE_SWAPPER?.addEventListener('click', swapShopAndUpgrade);
+  SHOP_UPGRADE_SWAPPER?.addEventListener('change', () => swapShopAndUpgrade());
```
Comment (Ido-Barnea) [inline core/development/ui/Events.ts:55]: Why not ` SHOP_UPGRADE_SWAPPER?.addEventListener('change', swapShopAndUpgrade);`?
Also, please remove the commented code.

File: core/styles/components.css [hunk 2]
```diff
@@ -214,3 +214,44 @@
...
+  height: 12px;
+  border-radius: 50%;
+  background: red;
+  position: absolute;
+  transition: 0.3s;
+}
+
+input[type='checkbox']:checked::before {
+  transform: translateX(60px);
+}
```
Comment (Ido-Barnea) [inline core/styles/components.css:247]: Please make the toggle button about half its current width. Also, if possible `margins`, `width`, and `height` should be relative to their parent (use `%`).

File: core/views/room.html [hunk 3]
```diff
@@ -13,9 +13,16 @@
   <body>
     <div id="left-column">
       <p id="info-container"></p>
-      <button id="shop-upgrade-swapper" class="center-horizontally">
+      <!-- <button id="shop-upgrade-swapper" class="center-horizontally">
         Switch Shop/Upgrade Tree
-      </button>
+      </button> -->
```
Comment (Ido-Barnea) [inline core/views/room.html:18]: Remove commented code

File: core/views/room.html [hunk 4]
```diff
@@ -13,9 +13,16 @@
...
+
+      <!-- Added toggle switch -->
+      <div class="checkbox-container">
+        <p >Switch Shop</p>
```
Comment (Ido-Barnea) [inline core/views/room.html:22]: I think it should be "Shop"

**Review by Ido-Barnea — CHANGES_REQUESTED** [review]
I see this is your first contribution here so welcome to Chess But Better! 🥳 🍾 
I only have a few minor requests before we can approve your PR, but other than that it looks great, glad to have you on the team 😄

**Comment by Ido-Barnea** [comment]
@SFarazH One last thing, please format your code using `npm run format`

File: core/styles/components.css (hunk 2 above)
Comment (SFarazH) [reply core/styles/components.css:247]: ok i have changed width
I tried to implement percentage wherever it was possible without altering the design too much

File: core/views/room.html (hunk 4 above)
Comment (SFarazH) [reply core/views/room.html:22]: changed

**Comment by SFarazH** [comment]
Made the changes along with running `npm run format`

File: core/styles/components.css [hunk 5]
```diff
@@ -214,3 +214,44 @@
   width: 100%;
   text-align: center;
 }
+
+/* toggle button */
+
+input[type='checkbox'] {
+  position: relative;
+  width: 40px;
+  height: 20px;
```
Comment (Ido-Barnea) [inline core/styles/components.css:223]: try changing it to `width: 6vh;` and `height: 3vh;`. Should look the same and be responsive.

File: core/styles/components.css [hunk 6]
```diff
@@ -214,3 +214,44 @@
...
+input[type='checkbox']::before {
+  content: '';
+  margin: 2.5px;
+  width: 12px;
+  height: 12px;
+  border-radius: 50%;
+  background: red;
+  position: absolute;
+  transition: 0.3s;
+}
```
Comment (Ido-Barnea) [inline core/styles/components.css:241]: Try the following:
- remove the `[margin](margin: 2.5px;)`
- set `width: 30%;` and `height: 70%;`
- set `top: 15%;` and `left: 10%;`

File: core/styles/components.css [hunk 7]
```diff
@@ -214,3 +214,44 @@
...
+input[type='checkbox'] {
+  position: relative;
+  width: 40px;
+  height: 20px;
+  appearance: none;
+  border-radius: 2rem;
+  cursor: pointer;
+  color: red;
+  border: 2px black solid;
+  padding-right: 4px;
```
Comment (Ido-Barnea) [inline core/styles/components.css:229]: No need for `padding-right: 4px;`

File: core/styles/components.css [hunk 8]
```diff
@@ -214,3 +214,44 @@
...
+  width: 12px;
+  height: 12px;
+  border-radius: 50%;
+  background: red;
+  position: absolute;
+  transition: 0.3s;
+}
+
+input[type='checkbox']:checked::before {
+  transform: translateX(175%);
```
Comment (Ido-Barnea) [inline core/styles/components.css:244]: To make the previous suggested changes work, set to `transform: translateX(165%);`

File: core/styles/components.css [hunk 9]
```diff
@@ -214,3 +214,44 @@
...
+}
+.checkbox-container {
+  display: flex;
+  margin: 7px;
```
Comment (Ido-Barnea) [inline core/styles/components.css:248]: replace `margin: 7px;` with `flex-direction: row;` and add `justify-content: center;`

**Review by Ido-Barnea — CHANGES_REQUESTED** [review]
I suggested some changes that will make your changes more responsive, please take a look

File: core/styles/components.css (hunk 5 above)
Comment (SFarazH) [reply core/styles/components.css:223]: done

**Comment by SFarazH** [comment]
> I suggested some changes that will make your changes more responsive, please take a look

Thanks for the recommendations.
I have applied the required changes.

**Review by Ido-Barnea — APPROVED** [review]
LGTM 🚀 👍🏼

**Comment by Monkey-bored** [comment]
Great work! 🎉

**Pull request merged**